assets/karma.conf.js

```diff
@@ -75,20 +75,91 @@ module.exports = function (config) {
         // enable / disable watching file and executing tests whenever any file changes
         autoWatch: false,
 
-
         // start these browsers
         // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
         browsers: ['Firefox'],
 
-
         // Continuous Integration mode
         // if true, Karma captures browsers, runs the tests and exits
         singleRun: true,
 
         // Concurrency level
         // how many browser should be started simultaneous
         concurrency: Infinity
-    });
+    };
+
+    // Add BrowserStack-specific settings
+    if (process.env.KARMA_BROWSERSTACK && process.env.BROWSER_STACK_USERNAME) {
```

Comment:
How does BROWSER_STACK_USERNAME get set?

Reply:
Via the secret environment variables in .travisci (or set locally)

Reply:
I don't understand ... do you have some links that could explain what's going on?

Reply:
Sure, here are the docs on that: https://docs.travis-ci.com/user/environment-variables/#Defining-encrypted-variables-in-.travis.yml

And a discussion of the problems of using it via forks: https://github.com/travis-ci/travis-ci/issues/1946

Reply:
I think I found it. It's in the .travis file. I got confused by the .travisci

Reply:
Oops, sorry 😄